presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -335,7 +362,10 @@ public static Slice charSubstr(@SqlType("char(x)") Slice utf8, @SqlType(Standard
     @ScalarFunction
     @LiteralParameters("x")
     @SqlType("varchar(x)")
-    public static Slice substr(@SqlType("varchar(x)") Slice utf8, @SqlType(StandardTypes.BIGINT) long start, @SqlType(StandardTypes.BIGINT) long length)
+    public static Slice substr(
+            @ScalarPropagateSourceStats(propagateAllStats = true) @SqlType("varchar(x)") Slice utf8,
```

Comment:
`substr` appears in almost all queries which have `LIKE %X` in them, for comparing/filtering on strings. Since it is so commonly used, shall we do a special handling of this function.

We could have a `estimate_ndv` function which takes the charset or character subset as argument and target length. For example if the charset subset is `BASIC_LATIN` [wiki link](https://en.wikipedia.org/wiki/Basic_Latin_(Unicode_block)) then we know NDV cannot be greater than 128.

One of the implementation challenge is - There is no stats that tells whether the column is `BASIC_LATIN` currently.